Megaparsec/Err.lean

```diff
@@ -25,6 +21,7 @@ def withHints (hs : Hints β)
               (f : Err m β ℘ E ξ) : Err m β ℘ E ξ :=
   fun e => match e with
   | .trivial pos us hs₀ =>
-      let hs' := unitMap $ List.join hs
-      f $ .trivial pos us $ hs₀ ++ hs'
+      -- we insert the given hints into `hs₀`
+      let hs' := (List.join hs).foldl .insert hs₀
+      f $ .trivial pos us hs'
```

Comment:
I don't understand this change.

Reply:
It would be cool to have it explained in the following way:

_Before we flattened hints and appended those to hs\_0. It was wrong because REASON, so now we flatten it and for each element in the flattened version, we ACTION it ADVERB hs\_0_.